Robot2019/src/main/java/frc/robot/RobotMap.java

```diff
@@ -18,28 +24,57 @@
  */
 public class RobotMap {
   static WPI_TalonSRX leftMaster, rightMaster;
+
   static WPI_VictorSPX leftSlave1, leftSlave2, rightSlave1, rightSlave2;
+
+  static Encoder leftEnc, rightEnc;
+
   static String driveMode;
 
+  static ArrayList<WPI_TalonSRX> masters;
+  static ArrayList<WPI_VictorSPX> slaves;
+  static ArrayList<Encoder> encoders;
+
   static {
-    // TODO: Put ports
     // Initialize motors on the left side of the drivetrain.
-    leftMaster = new WPI_TalonSRX(-1);
-    leftSlave1 = new WPI_VictorSPX(-1);
-    leftSlave2 = new WPI_VictorSPX(-1);
+    leftMaster = new WPI_TalonSRX(getPort("LeftMaster"));
+    leftSlave1 = new WPI_VictorSPX(getPort("LeftSlave1"));
+    leftSlave2 = new WPI_VictorSPX(getPort("LeftSlave2"));
 
     // Initialize motors on the right side of the drivetrain.
-    rightMaster = new WPI_TalonSRX(-1);
-    rightSlave1 = new WPI_VictorSPX(-1);
-    rightSlave2 = new WPI_VictorSPX(-1);
+    rightMaster = new WPI_TalonSRX(getPort("RightMaster"));
+    rightSlave1 = new WPI_VictorSPX(getPort("RightSlave1"));
+    rightSlave2 = new WPI_VictorSPX(getPort("RightSlave2"));
 
     // Slave the Victors to the Talons.
     leftSlave1.follow(leftMaster);
     leftSlave2.follow(leftMaster);
     rightSlave1.follow(rightMaster);
     rightSlave2.follow(rightMaster);
 
-    // Either arcade or tank
-    driveMode = "arcade";
+    masters.add(leftMaster);
+    masters.add(rightMaster);
+    slaves.add(leftSlave1);
+    slaves.add(leftSlave2);
+    slaves.add(rightSlave1);
+    slaves.add(rightSlave2);
+
+    leftEnc = new Encoder(new DigitalInput(getPort("LeftEncoder1")), new DigitalInput(getPort("LeftEncoder2")));
+    rightEnc = new Encoder(new DigitalInput(getPort("rightEncoder1")), new DigitalInput(getPort("rightEncoder2")));
```

Comment:
Historically we use the Encoder(int, int) constructor. Is there a reason for using this?

Reply:
Actually we've just been creating DigitalInput objects separately and the passing them into the Encoder objects. See this: https://github.com/DeepBlueRobotics/RobotCode2018/blob/master/Robot2018/src/main/java/org/usfirst/frc/team199/Robot2018/RobotMap.java#L154